### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by ibr5500**
## In this [PR] I have:

> - Add validation for the following ( name, posts_counter ) in User model.
> - Add validation for the following ( title, comments_counter, likes_counter) in Post model.
> - Add tests for the all models (User, Post, Comment, and Like).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -175,6 +177,23 @@ GEM
     reline (0.3.1)
       io-console (~> 0.5)
     rexml (3.2.5)
+    rspec-core (3.11.0)
+      rspec-support (~> 3.11.0)
+    rspec-expectations (3.11.0)
+      diff-lcs (>= 1.2.0, < 2.0)
+      rspec-support (~> 3.11.0)
+    rspec-mocks (3.11.1)
```
Comment (MeqdamAlqudah) [inline Gemfile.lock:185]: - [ ] Please add a test for validation in the user test I see that you just created an empty one 
image
it should look like this 
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @ibr5500 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:+1: Readme File


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile.lock (hunk 1 above)
Comment (ibr5500) [reply Gemfile.lock:185]: Sir @MeqdamAlqudah, I believe that what you suggest is optional for this project and not required.

Comment (ibr5500) [reply Gemfile.lock:185]: Hello, @MeqdamAlqudah  I believe that this is optional, in the project requirements there is only: Add unit specs for all of your models' methods and validations.

**Review by AshakaE — APPROVED** [review]
Hi @ibr5500 🙋‍♂️ , 

### Highlights ✨ 

- Thanks for following the requirements 👍 
- Pull request 👌 
- Git flow ✅ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ibr5500** [comment]
Thank you @AshakaE for the code review.
